ifrs17/Report/ReportConfigurationAndUtils.ipynb

```diff
@@ -183,6 +183,47 @@
             "execution_count": 0,
             "outputs": []
         },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Mutable Reporting Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static string ParseReportingPeriodToDisplayString(int year, int periodOfYear, char separator) => $\"{year} {separator}{periodOfYear}\";"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static string ParseDimensionToDisplayString(string systemName, string displayName) => $\"{displayName} ({systemName})\";"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<(IDictionary<string, string>, IReadOnlyCollection<string>)> GetAutocompleteMappings<T> (this IQuerySource querySource, bool hasNullAsFirstValue = default) where T : KeyedDimension {",
+                "\n    var query = await querySource.Query<T>().Select(x => new { x.SystemName, GuiName = ParseDimensionToDisplayString(x.SystemName, x.DisplayName), Order = 0 }).ToArrayAsync();",
+                "\n    var mappingDictionary = query.ToDictionary(x => x.GuiName, x => x.SystemName);",
+                "\n    var orderedDropDownValues = query.OrderBy(x => x.Order).ThenBy(x => x.GuiName).Select(x => x.GuiName);",
```

Comment:
You OrderBy x.Order here but in line 217 you set Order to 0. 
Is this OrderBy doing anything?

Reply:
It's a placeholder. We need to order by DisplayOrder if this is an ordered dimension, or in case of Hierarchical Dimension, the hierarchy. We need to write the specifics for these cases.

I think Katya should also provide this automatically from the hierarchical helper.